connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/PlanGenerationTestSuite.scala

```diff
@@ -2065,6 +2065,55 @@ class PlanGenerationTestSuite
       fn.lit(Array(new CalendarInterval(2, 20, 100L), new CalendarInterval(2, 21, 200L))))
   }
 
+  test("function typedLit") {
```

Comment:
Local checked: the new case can't pass with Scala 2.13

Reply:
How to recurrent ?

Reply:
```
gh pr checkout 40355
dev/change-scala-version.sh 2.13
build/sbt "connect-client-jvm/testOnly org.apache.spark.sql.PlanGenerationTestSuite" -Pscala-2.13

```

Reply:
run test with `SPARK_GENERATE_GOLDEN_FILES=1`, the diff as follows:

<img width="813" alt="image" src="https://user-images.githubusercontent.com/1475305/225195564-4d4907da-8e03-4971-bcb3-9175c4433878.png">

Reply:
I tested this case on my local machine. It's should be fixed.

Reply:
@beliefer how complete is this test? In other words, do you know of anything we do not support?

Reply:
Yeah that is fine.